### PR TITLE
Start To Work On Publisher Permission Creation

### PR DESCRIPTION
**Pull request by GuenoleCHERUBIN**
If publisher permission doesn't exist create it
Otherwize update it
Created new function CreateSocialMediaPermission

### CODE REVIEW

File: controllers/ManagementController.php [hunk 1]
```diff
@@ -61,7 +61,12 @@ private function actionModifiyAccount($id)
         }
 
         if ($formSocialMediaPermission->load($_POST) && $formSocialMediaPermission->validate()) {
-            $this->updateSocialMediaPermission($formSocialMediaPermission, $userPermission);
+            if ($userPermission == null){
```
Comment (EmileCalixte) [inline controllers/ManagementController.php]: You should use ===

Comment (EmileCalixte) [reply controllers/ManagementController.php]: Also, you should add a space between ) and {

File: controllers/ManagementController.php [hunk 2]
```diff
@@ -140,7 +165,7 @@ private function checkIfPermissionExists($id)
         if (!is_null($userPermission = SocialMediaPermission::findOne(['publisher_id' => $id]))) {
             return $userPermission;
         }
-        return null;
+        return $userPermission;
     }
```
Comment (EmileCalixte) [inline controllers/ManagementController.php]: This function is useless, it always returns the result of SocialMediaPermission::findOne

File: controllers/ManagementController.php [hunk 3]
```diff
@@ -140,7 +165,7 @@ private function checkIfPermissionExists($id)
         if (!is_null($userPermission = SocialMediaPermission::findOne(['publisher_id' => $id]))) {
             return $userPermission;
         }
-        return null;
+        return $userPermission;
```
Comment (alexisljn) [inline controllers/ManagementController.php]: You always returns $userPermission

**Review by alexisljn — CHANGES_REQUESTED** [review]

**Review by EmileCalixte — CHANGES_REQUESTED** [review]

**Review by EmileCalixte — APPROVED** [review]

**Review by alexisljn — APPROVED** [review]

**Pull request merged**